src/nspawn/nspawn-cgroup.c

```diff
@@ -100,14 +111,23 @@ int sync_cgroup(pid_t pid, CGroupUnified unified_requested) {
 
         undo_mount = true;
 
+        fn = strjoina(tree, cgroup);
+        (void) rm_rf(fn, REMOVE_ROOT|REMOVE_ONLY_DIRECTORIES);
```

Comment:
oh, i see now, this is because in this case PID 1 on the host of course won't take care of the other hierarchy, i see.